#25 del user

NKPmedia opened the pull request:
Pull request for

- #10 userList
- #12 addUser
- #17 userEdit
- #25 delUser???

Commit:
Fills the user roles with the groups the user is in (as ROLE_Groupname)

Commit:
Added a showAllUser template
Added a showAllUser page
Added function to UserRepo to get all Users
- You can filter them  by using a Filter Object

Commit:
Added a page to add Users
Tests need to be done

Commit:
Added tests for addUser
Added tests for ArrayMethods
Added a script to clear the ldap database
Phpunit will run this script before each test

Commit:
Added password support :D

Commit:
Fixed codeclimate issues

Commit:
Fixed codeclimate issues

Commit:
Fixed codeclimate issues

Commit:
Forgot a rename

Commit:
Added a page to show user details and added a function to edit the user

Commit:
Added a link to the userList page that links the user to the detail page of the user clicked on

Commit:
Fixed little test problem

Commit:
Added role hierarchy
Added more specific restriction rules to the detailUser page

Added a none elder user to the database

Commit:
Added testuser for the User remove function
replaced sudo with kdesudo

Commit:
Added new 403 errorPage

Added page to remove a user

NKPmedia committed:
Merge branch 'master' into #25-delUser

bone4 commented on app/Resources/translations/messages.de.yml:
```diff
@@ -1,6 +1,39 @@
 welcome: Willkommen
 login.request: Bitte logge dich ein
 username: Benutzername
+uidNumber: Benutzer ID
```
Aufräumen folgt mit #37

NKPmedia replied:
jo

bone4 commented on app/Resources/views/Exception/error403.html.twig:
```diff
@@ -0,0 +1,16 @@
+        <h1>You are not allowed to do this! ERROR 403</h1>
+
+        {% if is_granted('IS_AUTHENTICATED_FULLY') %}
+            <h2>Ask the admin why you are not allowed to do that</h2>
```
Meint IS_AUTHENTICATED_FULLY, dass der Benutzer eingeloggt ist?
Vielleicht schon ein Hinweis auf #38

NKPmedia replied:
Genau das ist gemeint :) 

Wollen wir den Hinweis in einem neues Branch hinzufügen?

NKPmedia replied:
So hab das jetzt schon hinzugefügt

bone4 commented on app/Resources/views/Exception/error403.html.twig:
```diff
@@ -0,0 +1,16 @@
+
+        <p>
+            You are not allowed to do that!<br>
+            Login with the correct user first <a href="{{ path('login') }}">Login</a>.
```
Ist vorher nicht ein Logout nötig?

NKPmedia replied:
jo

bone4 commented on app/Resources/views/includes/base.html.twig:
```diff
@@ -12,14 +12,20 @@
         <!-- Optional theme -->
         <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">
```
Wollen wir die Includes (JS & CSS) langfristig ins Projekt aufnehmen? Oder von entfernten Server includieren?

NKPmedia replied:
Mir egal. Wenn wir sie über cdns einbinden ist es eben leichter updates zu bekommen

bone4 replied:
Die feste Version, hier 3.3.7, muss wie sonst die Datei geupdated werden. Wie siehst du das @jonathan-jaekel ?

bone4 commented on app/Resources/views/includes/base.html.twig:
```diff
@@ -12,14 +12,20 @@
     <body>
         {% block body %}{% endblock %}
-        {% block javascripts %}{% endblock %}
+        {% block javascripts %}
```
Was bringt der {% block javascripts %} TWIG Block? Kann der JS-Teil nicht hard in den Quelltext gecodet werden?

NKPmedia replied:
Nein, denn wenn andere Seiten noch Javascript hinzufügen wollen, können sie es durch diesen Block an dieser Stelle tun

bone4 replied:
Du bist gerade im Besitz des Tickets #38 . Ist das Feedback.js dort richtig eingebunden?

bone4 commented on app/Resources/views/includes/removeUserButtonWithConfirmDialog.html.twig:
```diff
@@ -0,0 +1,20 @@
+            </div>
+        </div>
+    </div>
+</div>
```
No newline at and of file

bone4 commented on app/Resources/views/includes/showFlashes.html.twig:
```diff
@@ -7,3 +7,23 @@
+            </div>
+        </div>
+    {% endif %}
+{% endfor %}
```
No newline at and of file

bone4 commented on app/Resources/views/userManagment/addUser.html.twig:
```diff
@@ -0,0 +1,97 @@
+
+{% block title %}{% trans %}menu.userManagment.addUser{% endtrans %}{% endblock %}
+
+{% block stylesheets %}
```
Ist der stylesheet Abschnitt notwendig?

bone4 commented on app/Resources/views/userManagment/addUser.html.twig:
```diff
@@ -0,0 +1,97 @@
+    {{ parent() }}
+    <script src="{{ asset('js/addUser.js') }}"></script>
+
+{% endblock %}
```
No newline at and of file

bone4 commented on app/Resources/views/userManagment/detailUser.html.twig:
```diff
@@ -0,0 +1,279 @@
+
+{% block title %}Teams{% endblock %}
+
+{% block stylesheets %}
```
Ist der stylesheet Abschnitt notwendig?

bone4 commented on app/Resources/views/userManagment/detailUser.html.twig:
```diff
@@ -0,0 +1,279 @@
+{% block javascripts %}
+    {{ parent() }}
+
+{% endblock %}
```
No newline at and of file

bone4 commented on app/Resources/views/userManagment/detailUser.html.twig:
```diff
@@ -0,0 +1,279 @@
+
+{% endblock %}
+
+{% block javascripts %}
```
Muss Javascript hier nochmal eingebunden werden?

bone4 commented on app/Resources/views/userManagment/showAllUsers.html.twig:
```diff
@@ -0,0 +1,61 @@
+
+{% block title %}Alle Benutzer{% endblock %}
+
+{% block stylesheets %}
```
Ist der stylesheet Abschnitt hier notwendig?

bone4 commented on app/Resources/views/userManagment/showAllUsers.html.twig:
```diff
@@ -0,0 +1,61 @@
+        </table>
+    </div>
+
+{% endblock %}
```
No newline at and of file

bone4 commented on ldap-test-database/clear-ldap-database.sh:
```diff
@@ -0,0 +1,17 @@
+sudo slapadd -n 1 -l ./ldap-test-database/ldap-test-database-data.ldif
+sudo chown openldap:openldap /etc/ldap/slapd.d/ -R
+sudo chown openldap:openldap /var/lib/ldap/ -R
+sudo service slapd start
```
No newline at and of file

bone4 commented on src/AppBundle/Controller/UserManagment/UserController.php:
```diff
@@ -0,0 +1,268 @@
+    {
+        //Create the form
+        $userRepo = $this->get("data.userRepository");
+        $staemme = ["Ambronen","Hagen von Tronje","Anduril"];
```
Sollen die Stämme hier fest im Code stehen? Ist es möglich diese in eine Config auszulagern?

NKPmedia replied:
Nein sollen sie eigentlich nicht! Ich habe das bist jetzt nur so gemacht, weil ich mir noch nicht sicher war, wie man es sonst am besten macht:
- Config-Datei
- Datenbank
- Mit Hilfe des LDAP auslesen

bone4 commented on src/AppBundle/Controller/UserManagment/UserController.php:
```diff
@@ -0,0 +1,268 @@
+            } catch (UserAlreadyExistException $e) {
+                $this->addFlash("error", $e->getMessage());
+            } catch (ContextErrorException $e) {
+                $this->addFlash("error", $e->getMessage()." This probably means that this stamm (ou) does not exist.");
```
Text auf Deutsch übersetzen!?

bone4 commented on src/AppBundle/Controller/UserManagment/UserController.php:
```diff
@@ -0,0 +1,268 @@
+        $editUserForm = false;
+        //is the user allowed to edit this user?
+        //TODO add the ability for groupleaders to edit their users
+        if ($userToShow->getUid() == $loggedInUser->getUid()
```
Dieses lange if Statement ist nicht einfach überschaubar. Abfrage in eine eigene Methode auslagern!?

bone4 commented on src/AppBundle/Controller/UserManagment/UserController.php:
```diff
@@ -0,0 +1,268 @@
+            $userRepo = $this->get("data.userRepository");
+            $userToRemove = $userRepo->getUserByUid($uid);
+
+            if ($this->get('security.authorization_checker')->isGranted('ROLE_REMOVE_ALL_USERS')
```
gleiches if Statement, wie bei editUser?

bone4 commented on src/AppBundle/Controller/UserManagment/UserController.php:
```diff
@@ -0,0 +1,268 @@
+
+
+        } catch (UserDoesNotExistException $e) {
+            $this->addFlash("error", "The user $uid does not exist!");
```
deutscher Text

bone4 commented on src/AppBundle/Controller/UserManagment/UserController.php:
```diff
@@ -0,0 +1,268 @@
+        } catch (UserDoesNotExistException $e) {
+            $this->addFlash("error", "The user $uid does not exist!");
+        } catch (UserNotUniqueException $e) {
+            $this->addFlash("error", "The user $uid is not unique!");
```
deutscher Text

bone4 commented on src/AppBundle/Model/Services/GroupNotFoundException.php:
```diff
@@ -0,0 +1,16 @@
+<?php
+/**
```
Den Kommentar raus nehmen

bone4 commented on src/AppBundle/Model/User.php:
```diff
@@ -137,22 +163,52 @@ public function __construct($username, $password, $salt, array $roles)
+    public function setUid($uid)
     {
-        $this->username = $username;
+        $unwanted_array = array(    'Š'=>'S', 'š'=>'s', 'Ž'=>'Z', 'ž'=>'z', 'À'=>'A', 'Á'=>'A', 'Â'=>'A', 'Ã'=>'A',
```
Sind das alle?

bone4 commented on tests/AppBundle/Controller/UserControllerTest.php:
```diff
@@ -0,0 +1,168 @@
+
+        $this->assertContains('givenName=TestTronjer,ou=Hagen von Tronje,ou=People,dc=pbnl,dc=de', TestTools::getLoggedInStavoAmbrone()->getResponse()->getContent());
+    }
+}
```
No newline at and of file

bone4 replied:
Test zu delUser Id nicht vorhanden

bone4 replied:
Tests zu AddUser UserAlreadyExistException & ContextErrorException
(s. catch src/AppBundle/Controller/UserManagment/UserController.php line 131ff.)

bone4 commented on tests/AppBundle/ArrayMethodsTest.php:
```diff
@@ -0,0 +1,19 @@
+
+        $this->assertEquals($arrays, $array);
+    }
+}
```
No newline at and of file

bone4 commented on src/AppBundle/Security/AccessDeniedHandler.php:
```diff
@@ -0,0 +1,27 @@
+    {
+        return new Response($this->twig->render(':Exception:error403.html.twig',[]),403);
+    }
+}
```
No newline at and of file

bone4 commented on tests/AppBundle/FilterTest.php:
```diff
@@ -0,0 +1,19 @@
+        $this->assertEquals("test1", $filter->getFilterAttributes()[0]);
+        $this->assertEquals("test2", $filter->getFilterTexts()[0]);
+    }
+}
```
No newline at and of file

bone4 commented on tests/AppBundle/TestTools.php:
```diff
@@ -0,0 +1,42 @@
+            return TestTools::$loggedInStavoAmbrone;
+        }
+    }
+}
```
No newline at and of file

bone4 reviewed (commented):
Ein paar Anmerkungen hätte ich, siehe unten

Commit:
Added logout link to 403 error page
Added reference to feedback button to 403 error page

Commit:
Added translations for 403 error page

Commit:
Added more addUser test
Added more delUser tests
UserRepo->getUserByUid will now user findByUid in contrast to findOneByUid

Commit:
Merge branch '#25-delUser' of https://github.com/pbnl/ScoTooBo into #25-delUser

Commit:
Fixed codeclimate issues

Commit:
Fixed codeclimate issues

bone4 reviewed (approved):
Sieht für mich jetzt sehr gut aus! :+1:
